cmd/dataavailability/data_availability_check.go

```diff
@@ -0,0 +1,301 @@
+		if err != nil {
+			return err
+		}
+		for _, deliveredLog := range logs {
+			isDasMessage, err := d.checkDataAvailability(ctx, deliveredLog)
+			if err != nil {
+				return err
+			}
+			if isDasMessage {
+				return nil
+			}
+		}
+		latestBlock = latestBlock - d.config.L1BlocksPerRead
+	}
+	return fmt.Errorf("no das message found between block %d and block %d", latestBlock, oldBlock)
+}
+
+func (d *DataAvailabilityCheck) checkDataAvailabilityForOldHashInBlockRange(ctx context.Context, oldBlock uint64, latestBlock uint64) error {
+	for oldBlock+d.config.L1BlocksPerRead <= latestBlock {
+		query := ethereum.FilterQuery{
+			FromBlock: new(big.Int).SetUint64(oldBlock),
+			ToBlock:   new(big.Int).SetUint64(oldBlock + d.config.L1BlocksPerRead),
+			Addresses: []common.Address{*d.inboxAddr},
+			Topics:    [][]common.Hash{{batchDeliveredID}},
+		}
+		logs, err := d.l1Client.FilterLogs(ctx, query)
+		if err != nil {
+			return err
+		}
+		for _, deliveredLog := range logs {
+			isDasMessage, err := d.checkDataAvailability(ctx, deliveredLog)
+			if err != nil {
+				return err
+			}
+			if isDasMessage {
+				return nil
+			}
+		}
+		oldBlock = oldBlock + d.config.L1BlocksPerRead
+	}
+	return fmt.Errorf("no das message found between block %d and block %d", oldBlock, latestBlock)
+}
+
+// Trys to find if DAS message is present in the given log and if present
+// returns true and validates if the data is available in the storage service.
+func (d *DataAvailabilityCheck) checkDataAvailability(ctx context.Context, deliveredLog types.Log) (bool, error) {
+	data := []byte{}
+	deliveredEvent, err := d.inboxContract.ParseSequencerBatchDelivered(deliveredLog)
+	if err != nil {
+		return false, err
+	}
+	if deliveredEvent.DataLocation == uint8(batchDataSeparateEvent) {
+		query := ethereum.FilterQuery{
+			BlockHash: &deliveredLog.BlockHash,
+			Addresses: []common.Address{*d.inboxAddr},
+			Topics:    [][]common.Hash{{sequencerBatchDataABI.ID}, {common.BigToHash(deliveredEvent.BatchSequenceNumber)}},
+		}
+		logs, err := d.l1Client.FilterLogs(ctx, query)
+		if err != nil {
+			return false, err
+		}
+		if len(logs) != 1 {
+			return false, fmt.Errorf("found %d data logs for sequence 0x%x (expected 1)", len(logs), deliveredEvent.BatchSequenceNumber)
+		}
+		dataEvent, err := d.inboxContract.ParseSequencerBatchData(logs[0])
+		if err != nil {
+			return false, err
+		}
+		data = dataEvent.Data
+	} else if deliveredEvent.DataLocation == uint8(batchDataTxInput) {
+		txData, err := arbutil.GetLogEmitterTxData(ctx, d.l1Client, deliveredLog)
+		if err != nil {
+			return false, err
+		}
+		args := make(map[string]interface{})
+		err = addSequencerL2BatchFromOriginCallABI.Inputs.UnpackIntoMap(args, txData[4:])
+		if err != nil {
+			return false, err
+		}
+		var ok bool
+		data, ok = args["data"].([]byte)
+		if !ok {
+			return false, fmt.Errorf("couldn't parse data for sequence 0x%x", deliveredEvent.BatchSequenceNumber)
+		}
+	}
```

Comment:
This looks like the same code as what's in syncing_fallback_storage_service.go. Can you please find a way to reuse it?